game/scripts/ScenePool/ScenePool.gd

```diff
@@ -34,5 +34,4 @@ func acquire() -> BeepCube:
 	return cube
 
 func _on_scene_released(cube: BeepCube) -> void:
-	cube.scene_released.disconnect(_on_scene_released)
```

Comment:
@leandrodreamer haha i see you've been making attempts at improving cube cutting performance recently, and we have some conflicts.

I think the major issue with cubes leaking from the resource pool still exists on main (should probably retest though). I believe the big issue was that the release() signal was being disconnected here for no good reason, resulting in object not getting re-added to `_free_list` list after released(). This basically results in a pool object only ever getting used twice (i think) before the pool will become exhausted and start dynamically instancing entire new cube each time acquire() is called.

Reply:
Hm i see, i haven't noticed any performance issue on new cubes, even on web, but there may still be some issue if they stay in memory forever